amplitude.js

```diff
@@ -6047,34 +6080,44 @@ var _upgradeCookeData = function _upgradeCookeData(scope) {
  */
 var _loadCookieData = function _loadCookieData(scope) {
   var cookieData = scope.cookieStorage.get(scope.options.cookieName + scope._storageSuffix);
+
   if (type(cookieData) === 'object') {
-    if (cookieData.deviceId) {
-      scope.options.deviceId = cookieData.deviceId;
-    }
-    if (cookieData.userId) {
-      scope.options.userId = cookieData.userId;
-    }
-    if (cookieData.optOut !== null && cookieData.optOut !== undefined) {
-      scope.options.optOut = cookieData.optOut;
-    }
-    if (cookieData.sessionId) {
-      scope._sessionId = parseInt(cookieData.sessionId);
-    }
-    if (cookieData.lastEventTime) {
-      scope._lastEventTime = parseInt(cookieData.lastEventTime);
-    }
-    if (cookieData.eventId) {
-      scope._eventId = parseInt(cookieData.eventId);
-    }
-    if (cookieData.identifyId) {
-      scope._identifyId = parseInt(cookieData.identifyId);
-    }
-    if (cookieData.sequenceNumber) {
-      scope._sequenceNumber = parseInt(cookieData.sequenceNumber);
+    _loadCookieDataProps(scope, cookieData);
+  } else {
+    var legacyCookieData = scope.cookieStorage.get(scope.options.cookieName + scope._legacyStorageSuffix);
+    if (type(legacyCookieData) === 'object') {
+      _loadCookieDataProps(scope, legacyCookieData);
```

Comment:
Do you want to migrate / delete legacy cookie / save to new cookie?

Reply:
It does get saved. But there is no deletion here. Will add it.